### PR TITLE
bump c-ares/1.17.1

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **c-ares/1.17.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/c-ares/all/conanfile.py [hunk 1]
```diff
@@ -61,11 +62,14 @@ def package(self):
         tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
         tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
         tools.rmdir(os.path.join(self.package_folder, "share"))
+        for pdb_file in glob.glob(os.path.join(self.package_folder, "bin", "*.pdb")):
+            os.remove(pdb_file)
```
Comment (madebr) [inline recipes/c-ares/all/conanfile.py:66]:
```suggestion
        tools.remove_files_by_mask(os.path.join(self.package_folder, "bin"), "*.pdb")
```

File: recipes/c-ares/all/conanfile.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import glob
```
Comment (madebr) [inline recipes/c-ares/all/conanfile.py:1]:
```suggestion
```

File: recipes/c-ares/all/conanfile.py (hunk 1 above)
Comment (Croydon) [reply recipes/c-ares/all/conanfile.py:66]: Please set Conan minimum version to `1.29.1` when using this

**Review by prince-chrismc — DISMISSED** [review]

File: recipes/c-ares/all/conanfile.py (hunk 1 above)
Comment (prince-chrismc) [reply recipes/c-ares/all/conanfile.py:66]: another questions for #3614

File: recipes/c-ares/all/conanfile.py [hunk 3]
```diff
@@ -1,3 +1,4 @@
...
 import os
 from conans import ConanFile, CMake, tools
```
Comment (SpaceIm) [inline recipes/c-ares/all/conanfile.py:4]:
```suggestion

required_conan_version = ">=1.29.1"

```

**Review by prince-chrismc — APPROVED** [review]

**Review by ericLemanissier — APPROVED** [review]

**Review by intelligide — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**